### DISCUSSION TITLE
get all object paths with dot notation

### DISCUSSION THREAD

**Discussion by mesqueeb** [millsp/ts-toolbelt#286] [General]
I have a custom type that grabs all possible paths of an object with dot notation:

<img width="544" alt="image" src="https://user-images.githubusercontent.com/3253920/148671199-ecb14ca2-e6cf-4efa-aa78-7b2c61b10f1d.png">

The 4 is passed to make sure it only goes `4` levels deep. In order to keep the TS performance up.

I was wondering if this already existed in `ts-toolbelt`. I looked around but couldn't find it immediately!

I have made 2 really useful utilities for `OPaths` to get all paths, including branches and leaves, and `OLeaves` to just get the leaves.

My implementation is this:

https://www.typescriptlang.org/play?#code/PQKhCgAIUglBTALgVwE4DsDOlEAt44CeADgQGYD2qkmiqAlugOaQA29i8qAhq5gHRQI0SAHV86HPiKlI9bN0nwAtsUSEadRi3acerSAAMA5McOQqkbpCbx0XegGMjtBs3MdIqJGixH7AG5chkLA4OqyAJKYAMpazAAyHFy8ADwAKgB8kAC8kOmQ8AAenOgAJtiu2pAA-JAAFKaFJXYV+bWQZLyYBABcOKjI8ACUkP1dfPDg4KDCMABSFIzYiADuFJAA1vCE2IyIG9ZlFIiQ6CfciPQUksSXuEJCkHEUxAqn3Kg8u-yQAPJ4LireQETzrZCsMpydCOVjIMoEQwAEgA3uhkMoAEZcAC+IWgYQiBEWjFSAGkADSQAAK2TyZOapTaVWYUA61MZrUq8SYbMgHWicTcTCSejStM55RWgym-LlHWRKLJOP4qOpeL5cv6ZM1-Q5xSZ2HRWK4mrqOvl-UCpv5VvgQVQ02AwEgxG8iCuXAAtPQmOdvOESARqd4ArlIABta2oKkABipAEYqQAmKkAZipABYqQBWKkANipAHYqQAOKkATkT8cgCaTtdTtYzteztbztcLtZLtfLtarkGTNf4w9jEYAumOnWAngBBVgGYgUTCYeiY1gEP6YgBW8Ecp2p90wVMxPBh+GwM4AcgARNjwbhBTBPAACxW4qnX-wPeEwqRRVn6f9MStDFsWoHFIBxTInmdLwfAwbBjG4YxIAAH0gJD+ExYxQnAYpF1QU4iS-Q8MipW8DS5M5QK4cME1jOlI2vMdJTaKN7S4Sd+TqaM2X6ApKKlOBdyoMpUhZJgqUUQhoO4yAUU1fkIwZRgth2CgyHyMcvRqbVWO5YUFVRZVzHQkl0HJKk-m-XBf3SZSxypEN7QjZjMmyO0HU1HEI22QgNK0viMJwmZpxEOcFyXFc1w3bdd33Q9j2QU4blYDRATvB94CfERXyKd9iE-azSP-bhAMgYDqJNcDINkmBYPdXxEO4LCcIJPCigIoig3+BJ70fMjIAolohONMC6IY8NXJYwS2OjLiOlMIKBJGtoEEcUTxJ5KT0Bktk6gU+UlJUyQ-IC9JtN0yBBR5UUUlYclslm6Uhg6czLN6-rsoyBynNDab3LGYLvN89TNIuoKlsDWR0my048kOgD5Iq8rHHKsoQOqgBuQpMfGiCcRxxgZ36CTILZIpSZ5cc2UIcr0HfeAqeFcdIJp-kAC9ytBjRVIksdyo5oW8a4VmCbZDmE2TNNyu4KWZeRnnoU0YUBeRoWuaqsCxcg8mcWmfCqG62QADFWEuWHaBs7A8mKn8MjhrNoNgo6joAPRqA3OqNmQCDNi24b6rKbc+4OHdoJ2ZhdV35Q98AgA

**Comment by fullheart** [comment]
My (new) question is really similiar to yours @mesqueeb question: #296
